src/Collection/CollectionTrait.php

```diff
@@ -695,7 +695,10 @@ public function nest(
         };
 
         return $this->newCollection(new MapReduce($this->unwrap(), $mapper, $reducer))
-            ->map(fn ($value) => $isObject ? $value : $value->getArrayCopy());
+            ->map(function ($value) use ($isObject) {
```

Comment:
This looks good, but I wonder if static analyzers will let us annotate closure parameters on the line above the declaration.

Reply:
I tried, psalm doesnt like it.